src/sbl_filing_api/services/submission_processor.py

```diff
@@ -59,13 +60,27 @@ def validate_file_processable(file: UploadFile) -> None:
 
 async def upload_to_storage(period_code: str, lei: str, file_identifier: str, content: bytes, extension: str = "csv"):
     try:
-        fs: AbstractFileSystem = filesystem(settings.fs_upload_config.protocol)
-        if settings.fs_upload_config.mkdir:
+        if settings.fs_upload_config.protocol == FsProtocol.FILE:
+            fs: AbstractFileSystem = filesystem(settings.fs_upload_config.protocol)
             fs.mkdirs(f"{settings.fs_upload_config.root}/upload/{period_code}/{lei}", exist_ok=True)
-        with fs.open(
-            f"{settings.fs_upload_config.root}/upload/{period_code}/{lei}/{file_identifier}.{extension}", "wb"
-        ) as f:
-            f.write(content)
+            with fs.open(
+                f"{settings.fs_upload_config.root}/upload/{period_code}/{lei}/{file_identifier}.{extension}", "wb"
+            ) as f:
+                f.write(content)
+        else:
+            s3 = boto3.client("s3")
+            r = s3.put_object(
+                Bucket=settings.fs_upload_config.root,
+                Key=f"upload/{period_code}/{lei}/{file_identifier}.{extension}",
+                Body=content,
+            )
+            log.debug(
+                "s3 upload response for lei: %s, period: %s file: %s, response: %s",
+                lei,
+                period_code,
+                file_identifier,
+                r,
+            )
     except Exception as e:
         raise RegTechHttpException(
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR, name="Upload Failure", detail="Failed to upload file"
```

Comment:
I don't think this needs to be done here, but just realized it's a little weird throwing an HttpException here when for one of the uploads, it's done outside of an http request handler. It eventually gets caught as a general Exception and logged, but something to consider for naming purposes.

Reply:
yeah, this one is a bit weird, part of it does come from the request, i.e. uploading the submission file; then the report file gets uploaded outside of the request context; we can dig into how to present this better later, but for now think it's fine, since the fastapi exception handler would deal with the actual http exception, and the validation process has a try catch to deal with it.